test/zone.spec.js

```diff
@@ -11,6 +11,24 @@ describe('Zone', function () {
     expect(zone.fork().$id).toBeGreaterThan(zone.$id);
   });
 
+  it('should be possible to override the auto-generated id', function () {
+    zone.fork({'$id': 'test'}).run(function () {
+      expect(zone.$id).toEqual('test');
+    });
+  });
+
+  it('should return the full path', function () {
+    Zone.nextId = 2;
+
+    zone.fork().run(function () {
+      zone.fork({'$id': 'foo'}).run(function () {
```

Comment:
if you let people override $id, they could set it to 1, which could cause
all kinds of issues.

how about zone.fork([label], [options])?

so you would use it as:

zone.fork('myZone', {...})

I've also had a discussion with Misko about making custom zone installation
a different operation from forking, that would resolve this problem as well.

On Thu, May 21, 2015 at 11:36 PM Victor Berchet <notifications@github.com>
wrote:

> In test/zone.spec.js
> <https://github.com/angular/zone.js/pull/120#discussion_r30872178>:
>
> > @@ -11,6 +11,24 @@ describe('Zone', function () {
> >      expect(zone.fork().$id).toBeGreaterThan(zone.$id);
> >    });
> >
> > +  it('should be possible to override the auto-generated id', function () {
> > +    zone.fork({'$id': 'test'}).run(function () {
> > +      expect(zone.$id).toEqual('test');
> > +    });
> > +  });
> > +
> > +  it('should return the full path', function () {
> > +    Zone.nextId = 2;
> > +
> > +    zone.fork().run(function () {
> > +      zone.fork({'$id': 'foo'}).run(function () {
>
> I've pick $id because:
>
>    - we get the id via $id why not use the same mechanism to set it ?
>    - IMO the property name has to be "special", because you can set any
>    value on a zone and name is certainly too generic and it would be easy
>    to shoot yourself in a foot. I think it has to have some kind of
>    escaping/prefix
>
> Also $id is already kind of special, it changes for any new zone.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/zone.js/pull/120/files#r30872178>.
>